### PR TITLE
docs(private-cloud): standalone image updates

### PR DESCRIPTION
**Pull request by rarmatei**
Updates the docs for migrating to the newer `nxprivatecloud/single-image` standalone Docker image.

### CODE REVIEW

File: docs/nx-cloud/private/standalone.md [hunk 1]
```diff
@@ -14,22 +14,21 @@ documented. This document covers the latter version.
...
-When running everything together, you won't be able to run more than one instance of the Nx Cloud container. **So even
-though it is not required, we recommend you to run the MongoDB separately (see below how to do it).**
+You will then need to host the database separately:
 
-The instructions will go through running everything together first, and then, at the end, will talk about running the
-database and the file server separately.
+- Either by using a service such as Mongo Atlas
+- Or running it yourself. For that, we created the `nxprivatecloud/nx-cloud-mongo` image.
+
+The instructions will go through running the embeded file server, and then, at the end, will talk about hosting your cached artefacts on an external service, such as Amazon S3.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:24]:
```suggestion
The instructions will go through running the embedded file server, and then, at the end, will talk about hosting your cached artefacts on an external service, such as Amazon S3.
```

File: docs/nx-cloud/private/standalone.md [hunk 2]
```diff
@@ -219,7 +214,7 @@ creating the Nx Cloud docker container, like so:
 To obtain the `AZURE_CONNECTION_STRING` value go to your "Storage Account" and click on "Access Keys". You will also
 need to create a container in your storage account before starting the Nx Cloud container.
 
-If you use an external file storage and an external MongoDB instance, you don't have to provision the volume.
+If you use an external file storage, you don't have to provision the volume.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:217]:
```suggestion
If you use an external file storage solution, you don't have to provision the volume.
```

File: docs/nx-cloud/private/standalone.md [hunk 3]
```diff
@@ -280,3 +272,44 @@ Example:
 If you have a self-signed certificate, you will have to provision `NODE_EXTRA_CA_CERTS`. The env variable should point to a PEM file with either your certificate, or the root certificate your certificate was created from. Though this can be accomplished with a CLI command like `NODE_EXTRA_CA_CERTS=./tools/certs/cert.crt nx test myapp`, you will most likely want to configure it as a global env variable (for instance in your `.bashrc` file).
 
 A self-sign certificate registered in your OS won't be picked up by Node. Node requires you to provision `NODE_EXTRA_CA_CERTS`.
+
+## Migration guide from `nxprivatecloud/nxcloud` to `nxprivatecloud/single-image`
+
+If you are currently running `nxprivatecloud/nxcloud`, the older version of our Docker image, it is not being maintained anymore. This section will explain how to migrate to the new standalone image, `nxprivatecloud/single-image`.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:278]:
```suggestion
The old version of our Docker image, `nxprivatecloud/nxcloud`, is no longer being maintained. This section will explain how to migrate to the new standalone image, `nxprivatecloud/single-image`.
```

File: docs/nx-cloud/private/standalone.md [hunk 4]
```diff
@@ -280,3 +272,44 @@ Example:
...
+
+The main changes between the old image and the new one, is that the new one cannot run Mongo anymore. This makes for a much simpler image, based on Alpine (and not Ubuntu), with less chances for security vulnerabilities. It does require you to connect to an external Mongo instance, however. Below we'll cover the two possible migration scenarios.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:280]:
```suggestion
Unlike `nxprivatecloud/nx-cloud`, the `nxprivatecloud/single-image` container does not come with MongoDB included. This makes for a much simpler image, based on Alpine (and not Ubuntu), with less chances for security vulnerabilities. However, it does require you to connect to an external Mongo instance. Below we'll cover the two possible migration scenarios.
```

File: docs/nx-cloud/private/standalone.md [hunk 5]
```diff
@@ -280,3 +272,44 @@ Example:
...
+
+If you are currently relying on the image to host Mongo, we will need to move that to an external instance.
+
+##### Starting fresh with a dedicated Mongo instance
+
+1. We recommend setting up a dedicated Mongo host, on Atlas [as described above](#using-mongo-atlas). This means you will lose you current workspace set-up, but it is the easiest migration path and the most maintanable one.
+   1. To do this, get the connection string from Mongo Atlas
+   2. And configure the image with it: `-e NX_CLOUD_MONGO_SERVER_ENDPOINT="mongodb://domain-with-mongo:27017/nrwl-api"`
+   3. That's it!
+2. If you cannot use Atlas or Azure CosmosDB within your org, then you'll need to run our dedidcated Mongo Docker image. Read on.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:296]:
```suggestion
2. If you cannot use Atlas or Azure CosmosDB within your org, then you'll need to run our dedidcated Mongo Docker image. Instructions for this approach are described below.
```

File: docs/nx-cloud/private/standalone.md [hunk 6]
```diff
@@ -280,3 +272,44 @@ Example:
...
+
+##### Migrating your data to a separate self-hosted Mongo instance
+
+1. When running the image, you are probably mapping to the host volume like this `-v /data/private-cloud:/data:Z`
+2. That data folder is where the image stores all its persistent data that needs exist between restarts or updates of the image. Inside it you'll find:
+
+   1. Old image:
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:303]:
```suggestion
   1. In the old image (`nxprivatecloud/nx-cloud`):
```

File: docs/nx-cloud/private/standalone.md [hunk 7]
```diff
@@ -280,3 +272,44 @@ Example:
...
+      1. `/data/file-server`: if you are using the built-in file-server, this is where you'll find all the cached artefacts. If you are using an external S3 buckets this folder won't exist.
+      2. `/data/mongo`: this is where Mongo stores all its files.
+   2. In the new image:
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:306]:
```suggestion
   2. In the new image (`nxprivatecloud/single-image`):
```

**Review by bcabanes — APPROVED** [review]

File: docs/nx-cloud/private/standalone.md [hunk 8]
```diff
@@ -280,3 +272,44 @@ Example:
...
+      1. `/data/file-server` (if using the built-in file-server, otherwise you don't need to map anything)
+
+3. Copy the contents of `/data/mongo` folder on the host where you previously used to run Mongo, to the new host where you'll run the dedicated Mongo image
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:309]:
```suggestion
3. Copy the contents of `/data/mongo` folder from the host where you previously used to run Mongo, to the new host where you'll run the dedicated Mongo image
```

File: docs/nx-cloud/private/standalone.md [hunk 9]
```diff
@@ -14,22 +14,21 @@ documented. This document covers the latter version.
 2. MongoDB database
 3. File server
 
-By default, the container created by the `nxprivatecloud/nxcloud` image will create all three: the service, the
-database, and the file server. Using a single container is the easiest way to set it up, but it isn't the most robust
-way to run Nx Cloud.
+By default, the container created by the `nxprivatecloud/single-image` image will create two of them: the service, and the file server. Using a single container is the easiest way to set it up, but it isn't the most robust way to run Nx Cloud.
```
Comment (StalkAltan) [inline docs/nx-cloud/private/standalone.md:17]:
```suggestion
By default, the container created by the `nxprivatecloud/single-image` image will create two of them: the stateless Nx Cloud service, and the file server. Using a single container is the easiest way to set it up, but it isn't the most robust way to run Nx Cloud.
```

**Review by StalkAltan — COMMENTED** [review]
Some potential phrasing adjustments, content looks good! :+1:

**Pull request merged**